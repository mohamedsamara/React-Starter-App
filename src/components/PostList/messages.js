/*
 * PostList Messages
 *
 * This contains all the text for the PostList component.
 */
import { defineMessages } from 'react-intl';

export default defineMessages({
  header: {
    id: 'src.components.PostList.header',
    defaultMessage: 'This is the PostList component !',
  },
});
