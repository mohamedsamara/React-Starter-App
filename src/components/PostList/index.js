/**
 *
 * PostList
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';

import { withStyles } from '@material-ui/core/styles';
import ExpansionPanel from '@material-ui/core/ExpansionPanel';
import ExpansionPanelSummary from '@material-ui/core/ExpansionPanelSummary';
import ExpansionPanelDetails from '@material-ui/core/ExpansionPanelDetails';
import Typography from '@material-ui/core/Typography';
import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
import ExpansionPanelActions from '@material-ui/core/ExpansionPanelActions';
import Button from '@material-ui/core/Button';
import Divider from '@material-ui/core/Divider';
import DeleteIcon from '@material-ui/icons/Delete';

import messages from './messages';
import styles from './styles.css';

import { posts } from '../../containers/Post/types';

const muiStyles = theme => ({
  root: {
    width: '100%',
  },
  expanded: {
    '&:last-child': {
      marginBottom: 20,
    },
  },
  accordionSpace: {
    marginBottom: 20,
  },
  heading: {
    fontSize: theme.typography.pxToRem(15),
    fontWeight: theme.typography.fontWeightRegular,
  },
});

const PostList = props => {
  const { posts, classes, deletePost } = props;

  const postNodes = posts.map((post, index) => (
    <div key={post.id} className={classes.root} key={post.id}>
      <ExpansionPanel
        className={`${classes.accordionSpace} ${classes.expanded}`}
      >
        <ExpansionPanelSummary expandIcon={<ExpandMoreIcon />}>
          <Typography className={classes.heading}>{post.title}</Typography>
        </ExpansionPanelSummary>
        <ExpansionPanelDetails>
          <Typography>{post.body}</Typography>
        </ExpansionPanelDetails>

        <Divider />
        <ExpansionPanelActions>
          <Button size="small" color="primary">
            Save
          </Button>
          <Button
            color="primary"
            size="small"
            onClick={() => {
              deletePost(post.id, index);
            }}
          >
            Delete
            <DeleteIcon fontSize="small" />
          </Button>
        </ExpansionPanelActions>
      </ExpansionPanel>
    </div>
  ));

  return <div className={styles.postList}>{postNodes}</div>;
};

const { object, func } = PropTypes;
PostList.propTypes = {
  classes: object.isRequired,
  deletePost: func,
  posts: posts,
};

export default withStyles(muiStyles)(PostList);
