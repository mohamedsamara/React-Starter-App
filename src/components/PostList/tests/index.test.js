import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import PostList from '../index';

const posts = [
  {
    userId: 1,
    id: 1,
    title:
      'sunt aut facere repellat provident occaecati excepturi optio reprehenderit',
    body:
      'quia et suscipit\nsuscipit recusandae consequuntur expedita et cum\nreprehenderit molestiae ut ut quas totam\nnostrum rerum est autem sunt rem eveniet architecto',
  },
  {
    userId: 1,
    id: 2,
    title: 'qui est esse',
    body:
      'est rerum tempore vitae\nsequi sint nihil reprehenderit dolor beatae ea dolores neque\nfugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis\nqui aperiam non debitis possimus qui neque nisi nulla',
  },
  {
    userId: 1,
    id: 3,
    title: 'ea molestias quasi exercitationem repellat qui ipsa sit aut',
    body:
      'et iusto sed quo iure\nvoluptatem occaecati omnis eligendi aut ad\nvoluptatem doloribus vel accusantium quis pariatur\nmolestiae porro eius odio et labore et velit aut',
  },
];

const renderComponent = (props = {}) =>
  shallow(<PostList {...props} posts={posts} />);

describe('<PostList />', () => {
  it('should render an <PostList> tag', () => {
    const renderedComponent = renderComponent();

    expect(renderedComponent.exists()).toBe(true);
  });

  it('should <PostList> render correctly', () => {
    const component = renderer.create(<PostList posts={posts} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should have an posts attribute', () => {
    const renderedComponent = renderComponent();
    expect(renderedComponent.prop('posts')).toEqual(posts);
  });
});
