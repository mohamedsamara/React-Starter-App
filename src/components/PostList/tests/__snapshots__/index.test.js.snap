// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostList /> should <PostList> render correctly 1`] = `
<div
  className="postList"
>
  <div>
    <div
      className="PostList-root-1"
    >
      <div
        className="MuiPaper-root-9 MuiPaper-elevation1-12 MuiPaper-rounded-10 MuiExpansionPanel-root-5 MuiExpansionPanel-rounded-6 PostList-accordionSpace-3 PostList-expanded-2"
      >
        <div
          aria-expanded={false}
          className="MuiButtonBase-root-42 MuiExpansionPanelSummary-root-36"
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
        >
          <div
            className="MuiExpansionPanelSummary-content-40"
          >
            <p
              className="MuiTypography-root-45 MuiTypography-body1-54 PostList-heading-4"
            >
              sunt aut facere repellat provident occaecati excepturi optio reprehenderit
            </p>
          </div>
          <div
            aria-hidden="true"
            className="MuiButtonBase-root-42 MuiIconButton-root-81 MuiExpansionPanelSummary-expandIcon-41"
            onBlur={[Function]}
            onContextMenu={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex={-1}
          >
            <span
              className="MuiIconButton-label-86"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root-87"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                />
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
              </svg>
            </span>
            <span
              className="MuiTouchRipple-root-134"
            />
          </div>
        </div>
        <div
          aria-hidden="true"
          className="MuiCollapse-container-96"
          style={
            Object {
              "minHeight": "0px",
            }
          }
        >
          <div
            className="MuiCollapse-wrapper-98"
          >
            <div
              className="MuiCollapse-wrapperInner-99"
            >
              <div
                className="MuiExpansionPanelDetails-root-100"
              >
                <p
                  className="MuiTypography-root-45 MuiTypography-body1-54"
                >
                  quia et suscipit
suscipit recusandae consequuntur expedita et cum
reprehenderit molestiae ut ut quas totam
nostrum rerum est autem sunt rem eveniet architecto
                </p>
              </div>
              <hr
                className="MuiDivider-root-101"
              />
              <div
                className="MuiExpansionPanelActions-root-106"
              >
                <button
                  className="MuiButtonBase-root-42 MuiButton-root-108 MuiButton-text-110 MuiButton-textPrimary-111 MuiButton-flat-113 MuiButton-flatPrimary-114 MuiButton-sizeSmall-131 MuiExpansionPanelActions-action-107"
                  disabled={false}
                  onBlur={[Function]}
                  onContextMenu={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex="0"
                  type="button"
                >
                  <span
                    className="MuiButton-label-109"
                  >
                    Save
                  </span>
                  <span
                    className="MuiTouchRipple-root-134"
                  />
                </button>
                <button
                  className="MuiButtonBase-root-42 MuiButton-root-108 MuiButton-text-110 MuiButton-textPrimary-111 MuiButton-flat-113 MuiButton-flatPrimary-114 MuiButton-sizeSmall-131 MuiExpansionPanelActions-action-107"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onContextMenu={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex="0"
                  type="button"
                >
                  <span
                    className="MuiButton-label-109"
                  >
                    Delete
                    <svg
                      aria-hidden="true"
                      className="MuiSvgIcon-root-87 MuiSvgIcon-fontSizeSmall-94"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                      />
                      <path
                        d="M0 0h24v24H0z"
                        fill="none"
                      />
                    </svg>
                  </span>
                  <span
                    className="MuiTouchRipple-root-134"
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="PostList-root-1"
    >
      <div
        className="MuiPaper-root-9 MuiPaper-elevation1-12 MuiPaper-rounded-10 MuiExpansionPanel-root-5 MuiExpansionPanel-rounded-6 PostList-accordionSpace-3 PostList-expanded-2"
      >
        <div
          aria-expanded={false}
          className="MuiButtonBase-root-42 MuiExpansionPanelSummary-root-36"
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
        >
          <div
            className="MuiExpansionPanelSummary-content-40"
          >
            <p
              className="MuiTypography-root-45 MuiTypography-body1-54 PostList-heading-4"
            >
              qui est esse
            </p>
          </div>
          <div
            aria-hidden="true"
            className="MuiButtonBase-root-42 MuiIconButton-root-81 MuiExpansionPanelSummary-expandIcon-41"
            onBlur={[Function]}
            onContextMenu={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex={-1}
          >
            <span
              className="MuiIconButton-label-86"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root-87"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                />
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
              </svg>
            </span>
            <span
              className="MuiTouchRipple-root-134"
            />
          </div>
        </div>
        <div
          aria-hidden="true"
          className="MuiCollapse-container-96"
          style={
            Object {
              "minHeight": "0px",
            }
          }
        >
          <div
            className="MuiCollapse-wrapper-98"
          >
            <div
              className="MuiCollapse-wrapperInner-99"
            >
              <div
                className="MuiExpansionPanelDetails-root-100"
              >
                <p
                  className="MuiTypography-root-45 MuiTypography-body1-54"
                >
                  est rerum tempore vitae
sequi sint nihil reprehenderit dolor beatae ea dolores neque
fugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis
qui aperiam non debitis possimus qui neque nisi nulla
                </p>
              </div>
              <hr
                className="MuiDivider-root-101"
              />
              <div
                className="MuiExpansionPanelActions-root-106"
              >
                <button
                  className="MuiButtonBase-root-42 MuiButton-root-108 MuiButton-text-110 MuiButton-textPrimary-111 MuiButton-flat-113 MuiButton-flatPrimary-114 MuiButton-sizeSmall-131 MuiExpansionPanelActions-action-107"
                  disabled={false}
                  onBlur={[Function]}
                  onContextMenu={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex="0"
                  type="button"
                >
                  <span
                    className="MuiButton-label-109"
                  >
                    Save
                  </span>
                  <span
                    className="MuiTouchRipple-root-134"
                  />
                </button>
                <button
                  className="MuiButtonBase-root-42 MuiButton-root-108 MuiButton-text-110 MuiButton-textPrimary-111 MuiButton-flat-113 MuiButton-flatPrimary-114 MuiButton-sizeSmall-131 MuiExpansionPanelActions-action-107"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onContextMenu={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex="0"
                  type="button"
                >
                  <span
                    className="MuiButton-label-109"
                  >
                    Delete
                    <svg
                      aria-hidden="true"
                      className="MuiSvgIcon-root-87 MuiSvgIcon-fontSizeSmall-94"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                      />
                      <path
                        d="M0 0h24v24H0z"
                        fill="none"
                      />
                    </svg>
                  </span>
                  <span
                    className="MuiTouchRipple-root-134"
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="PostList-root-1"
    >
      <div
        className="MuiPaper-root-9 MuiPaper-elevation1-12 MuiPaper-rounded-10 MuiExpansionPanel-root-5 MuiExpansionPanel-rounded-6 PostList-accordionSpace-3 PostList-expanded-2"
      >
        <div
          aria-expanded={false}
          className="MuiButtonBase-root-42 MuiExpansionPanelSummary-root-36"
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
        >
          <div
            className="MuiExpansionPanelSummary-content-40"
          >
            <p
              className="MuiTypography-root-45 MuiTypography-body1-54 PostList-heading-4"
            >
              ea molestias quasi exercitationem repellat qui ipsa sit aut
            </p>
          </div>
          <div
            aria-hidden="true"
            className="MuiButtonBase-root-42 MuiIconButton-root-81 MuiExpansionPanelSummary-expandIcon-41"
            onBlur={[Function]}
            onContextMenu={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex={-1}
          >
            <span
              className="MuiIconButton-label-86"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root-87"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                />
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
              </svg>
            </span>
            <span
              className="MuiTouchRipple-root-134"
            />
          </div>
        </div>
        <div
          aria-hidden="true"
          className="MuiCollapse-container-96"
          style={
            Object {
              "minHeight": "0px",
            }
          }
        >
          <div
            className="MuiCollapse-wrapper-98"
          >
            <div
              className="MuiCollapse-wrapperInner-99"
            >
              <div
                className="MuiExpansionPanelDetails-root-100"
              >
                <p
                  className="MuiTypography-root-45 MuiTypography-body1-54"
                >
                  et iusto sed quo iure
voluptatem occaecati omnis eligendi aut ad
voluptatem doloribus vel accusantium quis pariatur
molestiae porro eius odio et labore et velit aut
                </p>
              </div>
              <hr
                className="MuiDivider-root-101"
              />
              <div
                className="MuiExpansionPanelActions-root-106"
              >
                <button
                  className="MuiButtonBase-root-42 MuiButton-root-108 MuiButton-text-110 MuiButton-textPrimary-111 MuiButton-flat-113 MuiButton-flatPrimary-114 MuiButton-sizeSmall-131 MuiExpansionPanelActions-action-107"
                  disabled={false}
                  onBlur={[Function]}
                  onContextMenu={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex="0"
                  type="button"
                >
                  <span
                    className="MuiButton-label-109"
                  >
                    Save
                  </span>
                  <span
                    className="MuiTouchRipple-root-134"
                  />
                </button>
                <button
                  className="MuiButtonBase-root-42 MuiButton-root-108 MuiButton-text-110 MuiButton-textPrimary-111 MuiButton-flat-113 MuiButton-flatPrimary-114 MuiButton-sizeSmall-131 MuiExpansionPanelActions-action-107"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onContextMenu={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex="0"
                  type="button"
                >
                  <span
                    className="MuiButton-label-109"
                  >
                    Delete
                    <svg
                      aria-hidden="true"
                      className="MuiSvgIcon-root-87 MuiSvgIcon-fontSizeSmall-94"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                      />
                      <path
                        d="M0 0h24v24H0z"
                        fill="none"
                      />
                    </svg>
                  </span>
                  <span
                    className="MuiTouchRipple-root-134"
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
